diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -210,6 +210,6 @@ include(CTest)          #           "
 
 set_target_processor_type(CPU_ARCHITECTURE) # Must be after project.
-if (CPU_ARCHITECTURE STREQUAL x86)
+if (CPU_ARCHITECTURE STREQUAL x86 AND NOT (ANDROID AND MU3D_ALLOW_ANDROID_X86))
     message(FATAL_ERROR "This project cannot be built for x86 cpu.")
 endif()
 CMAKE_DEPENDENT_OPTION( BASISU_SUPPORT_SSE
